pkg/client/typed/discovery/discovery_client.go

```diff
@@ -163,6 +171,50 @@ func (d *DiscoveryClient) ServerResources() (map[string]*unversioned.APIResource
 	return result, nil
 }
 
+// serverPreferredResources returns the supported resources with the version preferred by the
+// server. If namespaced is true, only namespaced resources will be returned.
+func (d *DiscoveryClient) serverPreferredResources(namespaced bool) ([]unversioned.GroupVersionResource, error) {
```

Comment:
> So does the garbage collector ignore the version specified in the owner reference?

No, gc uses the dynamic client and that client will use the version specified in the owner reference.

GC only uses the preferredVersion to watch the resource.

I don't see why third party resource will be a problem, could you explain more?